packages/base-controller/src/RestrictedMessenger.ts

```diff
@@ -409,6 +426,35 @@ export class RestrictedControllerMessenger<
    * @returns Whether the name is within the current namespace
    */
   #isInCurrentNamespace(name: string): name is NamespacedName<Namespace> {
-    return name.startsWith(`${this.#controllerName}:`);
+    return name.startsWith(`${this.#namespace}:`);
   }
 }
+
+/**
+ * Constructs a restricted messenger
+ *
+ * The provided allowlists grant the ability to call the listed actions and subscribe to the
+ * listed events. The "name" provided grants ownership of any actions and events under that
+ * namespace. Ownership allows registering actions and publishing events, as well as
+ * unregistering actions and clearing event subscriptions.
+ *
+ * @param options - Options.
+ * @param options.controllerMessenger - The messenger instance that is being wrapped. (deprecated)
+ * @param options.messenger - The messenger instance that is being wrapped.
+ * @param options.name - The name of the thing this messenger will be handed to (e.g. the
+ * controller name). This grants "ownership" of actions and events under this namespace to the
+ * restricted messenger returned.
+ * @param options.allowedActions - The list of actions that this restricted messenger should be
+ * allowed to call.
+ * @param options.allowedEvents - The list of events that this restricted messenger should be
+ * allowed to subscribe to.
+ * @deprecated This has been renamed to `RestrictedMessenger`.
+ */
+export const RestrictedControllerMessenger = RestrictedMessenger;
+export type RestrictedControllerMessenger<
+  Namespace extends string,
+  Action extends ActionConstraint,
+  Event extends EventConstraint,
+  AllowedAction extends string,
+  AllowedEvent extends string,
+> = RestrictedMessenger<Namespace, Action, Event, AllowedAction, AllowedEvent>;
```

Comment:
Hmm, this would not allow us to gradually switch from one to the other. But it does allow us to separate this rename PR from the later changes elsewhere. Still a win. Thanks for the suggestion

Reply:
Apologies if this is a silly question, but why wouldn’t this approach allow us to transition gradually?

Reply:
Oh nevermind, thanks for following up, I thought you were removing the aliases here. But you are not. I will do as you suggest across both packages.

Reply:
Done here: https://github.com/MetaMask/core/pull/5050/commits/1d772cc55a15884e7f5981ccf57cd59b3ba1094a

Seems to work! 🎉